### ISSUE TITLE
alwaysThrows should silence missing return hints

### ISSUE THREAD

**Issue by nex3** [dart-lang/sdk#31384]
When I run the analyzer (2.0.0-dev.7) on the following code:

```dart
import 'package:meta/meta.dart';

int foo(bool yes) {
  if (yes) return 1;
  error("no");
}

@alwaysThrows
void error(String message) {
  throw message;
}
```

I get this message:

```
hint • This function declares a return type of 'int', but doesn't end with a return statement at test.dart:3:1 • missing_return   
```

Because the analyzer knows that `error()` always throws, I'd expect it to be able to deduce that no return is needed, just like it does if I manually write `throw "no"`.

**Comment by nex3** [comment]
cc @srawlins, who added `@alwaysThrows` in c6d6a69a7876c2b3d637472f4f8594f18de82263.

**Comment by srawlins** [comment]
Fixed with 0a6253312f2263bb82ca7ef9642339a530f15493.
